packages/providers/upload-local/README.md

```diff
@@ -35,9 +35,7 @@ module.exports = ({ env }) => ({
   upload: {
     config: {
       provider: 'local',
-      providerOptions: {
-        sizeLimit: 100000,
-      },
+      sizeLimit: 100000,
```

Comment:
With this implementation, sizeLimit is not getting to the provider init, as it is not inside `providerOptions`.
This would be a breaking change.

Still, I think this is the way it should be done. And maybe we could allow sizeLimit both to be in `config` and `config.providerOptions`. And giving a deprecation warning for the second one.

Let me discuss this with @petersg83  this afternoon.  I will update you asap.

Reply:
Ok! So we discussed about this!
Our conclusion is pretty aligned I think:

- sizeLimit should be in `config` , not in `config.providerOptions`.
- This change would be breaking, so we should leave the two. If using it in the provider Options, we should print a warning like:
```js
// TODO V5: remove sizeLimit from providerOptions
process.emitWarning(
    '[deprecated] In future versions, `sizeLimit` argument will be ignored from 
     upload.config.providerOptions. Move it to  upload.config'
);
```
- Change the [docs](https://docs.strapi.io/developer-docs/latest/plugins/upload.html#configuration:~:text=can%20pass%20the-,sizeLimit,-%2C%20which%20is%20an) explaining the change.